ovos_PHAL_plugin_oauth/__init__.py

```diff
@@ -26,18 +30,29 @@ class OAuthPlugin(PHALPlugin):
 
     def __init__(self, bus=None, config=None):
         super().__init__(bus=bus, name="ovos-PHAL-plugin-oauth", config=config)
-        self.port = config.get("port", 36536)
-
+        self.config = config
+        self.port = self.config.get("port", 36536)
         self.bus.on("oauth.register", self.handle_oauth_register)
         self.bus.on("oauth.start", self.handle_start_oauth)
         self.bus.on("oauth.get", self.handle_get_auth_url)
-        self.bus.on("ovos.shell.oauth.register.credentials", self.handle_client_secret)
+        self.bus.on("ovos.shell.oauth.register.credentials",
+                    self.handle_client_secret)
+
+        # QR Code Remote OAuth Process Support
+        self.bus.on("oauth.get.app.host.info", self.handle_get_app_host_info)
+        self.bus.on("oauth.generate.qr.request", self.handle_generate_qr)
 
         # trigger register events from oauth skills
         self.bus.emit(Message("oauth.ping"))
+        self.local_flask_host = None
 
         self.oauth_skills = {}
 
+        # Run the flask server once all has actually initialized
+        # This is to prevent the flask server from starting before the plugin has been fully initialized
+        # otherwise self.port may not be set yet and the flask server will start on the default port
+        self.run_flask_app()
```

Comment:
this should be done inside the run method, the plugin itself is already a Thread

you can use a threading.Event instead and wait for it in the run method to be sure all init steps already happened